pkg/templates/kubeadm/v1beta3/kubeadm.go

```diff
@@ -375,10 +401,22 @@ func newNodeIP(host kubeoneapi.HostConfig) string {
 
 func newNodeRegistration(s *state.State, host kubeoneapi.HostConfig) kubeadmv1beta3.NodeRegistrationOptions {
 	kubeletCLIFlags := map[string]string{
-		"node-ip":           newNodeIP(host),
 		"volume-plugin-dir": "/var/lib/kubelet/volumeplugins",
 	}
 
+	// If external or in-tree CCM is in use we don't need to set --node-ip
+	// as the cloud provider will know what IPs to return.
+	if !s.Cluster.CloudProvider.External {
+		switch {
+		case s.Cluster.ClusterNetwork.IPFamily == kubeoneapi.IPFamilyIPv4IPv6:
+			kubeletCLIFlags["node-ip"] = newNodeIP(host) + "," + host.IPv6Addresses[0]
+		case s.Cluster.ClusterNetwork.IPFamily == kubeoneapi.IPFamilyIPv6IPv4:
+			kubeletCLIFlags["node-ip"] = host.IPv6Addresses[0] + "," + newNodeIP(host)
+		default:
+			kubeletCLIFlags["node-ip"] = newNodeIP(host)
+		}
+	}
```

Comment:
Are we sure that this works for all cloud providers and CCMs? What if we do this unconditionally to be safe?

Reply:
`--node-ip` and `--cloud-provider` kubelet flags conflict in the dualstack setup.

 In general, it is not expected to need to use `--node-ip` with external CCMs, as the the cloud provider is expected to know the correct IPs to return.

ref https://github.com/kubermatic/machine-controller/pull/1336

I should be checking for dualstack somewhere around there. :runner: